File: Linguagem de programacao III/lista1_python/CRAPs.py
# Um dos jogos de azar mais populares é o jogo conhecido como Craps. Escreva
# um programa em Java que simula o jogo de Craps.

#biblioteca para gerar números aleatorios
import random

#valores aleatorios atribuídos aos dois dados
dado1 = random.randrange(1, 7)
dado2 = random.randrange(1, 7)

#caso aconteça uma segunda rodada, guarda a soma dos dados
ponto = 0

print("dado 1: ", dado1, ", dado 2: ", dado2)

#se a soma for igual a 7 ou 11: venceu
if (dado1 + dado2) == 7 or (dado1 + dado2) == 11:
    print("voce ganhou :-)\n")

#se a soma for igual a 2, 3 ou 12: perdeu
elif (dado1 + dado2) == 2 or (dado1 + dado2) == 3 or (dado1 + dado2) == 12:
    print("voce perdeu :-(\n")

#caso contrário
else:
    #somas dos dados: critério para vitória
    ponto = dado1 + dado2

    #quantidades de jogadas até o jogo acabar
    control = 1

    print("ponto: ", ponto, "\nIniciando estagio 2\n")

    #loop que se encerra quando vence ou perde
    while True:
        #novos valores atribuidos aos dados a cada rodada
        dado1 = random.randrange(1, 7)
        dado2 = random.randrange(1, 7)

        print("dado 1: ", dado1, ", dado 2: ", dado2)
        print("jogada ", control, ": ", (dado1 + dado2))

        #condições de parada
        if (dado1 + dado2) == ponto:
            print("voce ganhou :-)\n")
            break
        elif (dado1 + dado2) == 7:
            print("voce perdeu :-(\n")
            break

        control += 1

File: Linguagem de programacao III/lista1/craps.java
/* Um dos jogos de azar mais populares é o jogo conhecido como Craps. Escreva
 * um programa em Java que simule o jogo de Craps.
 */

import java.util.Random;

public class craps {
  public static void main(String[] args) 
  {
    // função geradora de numeros aleatorios 
    Random random = new Random();

    // Variaveis que serão os dois dados do jogo
    int dado1, dado2;

    dado1 = random.nextInt(5) + 1; // gera um numero aleatorio de 1 a 6
    dado2 = random.nextInt(5) + 1; // e atribui ao dado 1 e 2

    // variavel que sera usado caso ocorra o segundo estágio
    int ponto; 

    System.out.println("Dado 1: " + dado1 + ", Dado 2: " + dado2); // print dados

    // Se soma for igual a 7 ou 11: ganhou
    if((dado1 + dado2) == 7 || (dado1 + dado2) == 11)
      System.out.println("Voce ganhou :-)");
    // se soma for igual a 2, 3 ou 12: perdeu
    else if((dado1 + dado2) == 2 || (dado1 + dado2) == 3 || (dado1 + dado2) == 12)
      System.out.println("Voce perdeu :-(");
    // caso contrario, entra no loop
    else 
    {
      //Atribuição do valor do ponto
      ponto = dado1 + dado2;
      int i = 1; // Contador de jogadas
      
      System.out.println("ponto: " + ponto); // print ponto
      System.out.println("iniciando estagio 2");

      //loop que se encerra ou quando vence ou perde
      while(true)
      {
        // Gerando novos valores aleatorios
        dado1 = random.nextInt(5) + 1;
        dado2 = random.nextInt(5) + 1;
        
        System.out.println("Dado 1: " + dado1 + ", Dado 2: " + dado2); // print dados
        System.out.println("Jogada " + i + ": " + (dado1 + dado2));

        // condiçoes de parada 
        if((dado1 + dado2) == ponto)
        {
          System.out.println("Voce ganhou :-)");
          break;
        }
        else if((dado1 + dado2) == 7)
        {
          System.out.println("Voce perdeu :-(");
          break;
        }

        i++; // incremento do contador
      }
    }
  }
}